### PR TITLE
Added InventoryManger, Auto Obsidian and some bug fixes

### PR DESCRIPTION
**Pull request by Xiaro**
**Describe the pull**
1. Added Inventory manger (issue #1038)
2. Add Auto Obsidian (Speical thanks to CorruptedSeal for help)
3. Better Baritone Pausing
4. Baritone GUI now shows all processes except farming (issue #1048)
5. Packet mine now sends stops STOP_DESTROY_BLOCK packet after a short delay
6. Add a limit to decimal places for DiscordRPC speed
7. Don't attack the entity you are riding on lol (issue #1041)

**Describe how this pull is helpful**

1. Refills items in hotbar, Save damaged tools/weapon and swap to a new one if possible, Ejects items you don't want.
2. Mines enderchest to get obsidian automatically, compatible with baritone. Helps with building giant obsidian structure
3. More kotlin and not cancelling other process
4. Just a bug fixes
5. So you don't have to release the left click to get it work, making it compatible with baritone
6. You don't need that many decimal places for DiscordRPC lol
7. Just a bug fixes

### CODE REVIEW

File: src/main/java/me/zeroeightsix/kami/KamiMod.java [hunk 1]
```diff
@@ -98,6 +99,7 @@
     public static boolean hasAskedToUpdate = false;
 
     public static TemporaryPauseProcess pauseProcess;
+    public static AutoObsidianProcess autoObsidianProcess;
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/KamiMod.java:102]: is there a good reason as to why this is in kamimod?

File: src/main/java/me/zeroeightsix/kami/command/commands/EjectCommand.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                sendChatMessage("""Available options: 
+  +item: Adds item to the list
+  -item: Removes item from the list
+  =item: Changes the list to only contain item
+  list: Prints the list of selected items
+  defaults: Resets the list to the default list
+  clear: Removes all items from the AutoEject item list""")
+            }
+
+            args[0]!!.startsWith("+", true) -> {
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/command/commands/EjectCommand.kt:41]: are you sure these should be `!!`? won't they be null if they don't exist?

File: src/main/java/me/zeroeightsix/kami/module/modules/misc/AutoObsidian.kt [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+
+    private fun lookAtBlock(pos: BlockPos) {
+        val vec3d = Vec3d((pos.x + 0.5) - mc.player.posX, pos.y - (mc.player.eyeHeight + mc.player.posY), (pos.z + 0.5) - mc.player.posZ)
+        val lookAt = getRotationFromVec3d(vec3d)
+        mc.player.rotationYaw = lookAt[0].toFloat()
+        mc.player.rotationPitch = lookAt[1].toFloat()
+    }
+
+    /* Tasks */
+    // TODO: SHULKERS SEARCHING
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/misc/AutoObsidian.kt:263]: uhhh is this finished or?

File: src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraFlight.kt [hunk 4]
```diff
@@ -318,35 +319,6 @@ class ElytraFlight : Module() {
...
-    private fun checkForLiquid(): Boolean {
-        return getGroundPosY(true) == -1.0f
-    }
-
-    /**
-     * Get the height of the ground surface below, and check for liquid if [checkLiquid] is true
-     *
-     * @return The y position of the ground surface, -1.0f if found liquid below and [checkLiquid] is true
-     */
-    private fun getGroundPosY(checkLiquid: Boolean): Float {
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraFlight.kt:333]: oh uhhhh the vec helper already had this function, oops

File: src/main/java/me/zeroeightsix/kami/module/modules/player/AutoEat.kt [hunk 5]
```diff
@@ -64,10 +63,8 @@ class AutoEat : Module() {
                 lastSlot = -1
             }
             eating = false
+            unpause()
 
-            if (pauseBaritone.value) {
-                unpause()
-            }
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/player/AutoEat.kt:70]: oops, my bad :p

File: src/main/java/me/zeroeightsix/kami/module/modules/player/InventoryManager.kt [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+import me.zeroeightsix.kami.util.InventoryUtils.throwAllInSlot
+import net.minecraft.client.gui.inventory.GuiContainer
+import net.minecraft.item.Item.getIdFromItem
+import net.minecraft.item.ItemStack
+import kotlin.math.ceil
+
+/**
+ * Created by Xiaro on 7/13/20
+ */
+
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/player/InventoryManager.kt:30]: remove this empty space pls

File: src/main/java/me/zeroeightsix/kami/process/AutoObsidianProcess.kt [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    override fun priority(): Double {
+        return 2.0
+    }
+
+    override fun onLostControl() {}
+
+    override fun displayName0(): String {
+        return "AutoObsidian: " + KamiMod.MODULE_MANAGER.getModuleT(AutoObsidian::class.java).state.toString().toLowerCase()
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/process/AutoObsidianProcess.kt:30]: maybe keep this in the UPPER_CASE (remove .toLowerCase())

File: src/main/java/me/zeroeightsix/kami/process/TemporaryPauseProcess.kt [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        }
+        return PathingCommand(null, PathingCommandType.REQUEST_PAUSE)
+    }
+
+    override fun onLostControl() {
+        // nothing :p
+    }
+
+    override fun displayName0(): String {
+        return "Kami Blue Pauser"
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/process/TemporaryPauseProcess.kt:46]: capital KAMI Blue

File: src/main/java/me/zeroeightsix/kami/module/modules/misc/AutoObsidian.kt (hunk 3 above)
Comment (Xiaro) [reply src/main/java/me/zeroeightsix/kami/module/modules/misc/AutoObsidian.kt:263]: Ehh yea I forgot to remove it lol

File: src/main/java/me/zeroeightsix/kami/command/commands/EjectCommand.kt (hunk 2 above)
Comment (Xiaro) [reply src/main/java/me/zeroeightsix/kami/command/commands/EjectCommand.kt:41]: Maybe I should put the non null check in an if statement not when statement

File: src/main/java/me/zeroeightsix/kami/KamiMod.java (hunk 1 above)
Comment (Xiaro) [reply src/main/java/me/zeroeightsix/kami/KamiMod.java:102]: Just to initialize the baritone process

File: src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraFlight.kt (hunk 4 above)
Comment (Xiaro) [reply src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraFlight.kt:333]: This one uses player bounding box to check the ground position

File: src/main/java/me/zeroeightsix/kami/process/AutoObsidianProcess.kt (hunk 7 above)
Comment (Xiaro) [reply src/main/java/me/zeroeightsix/kami/process/AutoObsidianProcess.kt:30]: It looks weird if you have full upper case word with normal words together

File: src/main/java/me/zeroeightsix/kami/command/commands/EjectCommand.kt (hunk 2 above)
Comment (Xiaro) [reply src/main/java/me/zeroeightsix/kami/command/commands/EjectCommand.kt:41]: There is a non null check above, see line 31

**Pull request merged**